build.sh

```diff
@@ -3,12 +3,23 @@
 rm -rf dist
 mkdir dist
 
-echo "Run dep ensure"
-$GOPATH/bin/dep ensure
-$GOPATH/bin/dep check
+# generate version number
+if [ -n "${TRAVIS_TAG:-}" ]; then
+    VERSION_BASE=${TRAVIS_TAG}  
+else 
+    VERSION_BASE=$(git describe --abbrev=0 --tags 2> /dev/null || echo 'dblp.0.0.0')
+fi
+VERSION_SHORT_COMMIT=$(git rev-parse --short HEAD)
+VERSION_FULL_COMMIT=$(git rev-parse HEAD)
+
+GO_COMMON_PATH="github.com/tidepool-org/go-common"
+	
+echo "Build hydrophone $VERSION_BASE+$VERSION_FULL_COMMIT"
+go build -ldflags "-X $GO_COMMON_PATH/clients/version.ReleaseNumber=$VERSION_BASE \
```

Comment:
Looks like a good practice to run "go mod tidy" to prune the dependencies before building. Does it make sense?
https://github.com/golang/go/wiki/Modules#how-to-prepare-for-a-release